powersimdata/input/transform_profile.py

```diff
@@ -117,6 +117,48 @@ def _get_demand_profile(self):
                 demand.loc[:, key] *= value
         return demand
 
+    def _get_demand_flexibility_profile(self, name):
+        """Return the appropriately pruned demand flexibility profiles. Provides support
+        for profiles that might have a mixed input of zones and buses.
+
+        :param string name: The type of demand flexibility profile being specified. Can
+            be one of: *'demand_flexibility_up'*, *'demand_flexibility_dn'*,
+            *'demand_flexibility_cost_up'*, or *'demand_flexibility_cost_dn'*.
+        :return: (*pandas.DataFrame*) -- data frame of a demand flexibility profile.
+        """
+        # Access the specified demand flexibility profile
+        flex_dem_dict = self.ct["demand_flexibility"]
+        flex_dem_dict["grid_model"] = self.scenario_info["grid_model"]
+        df = self._input_data.get_data(flex_dem_dict, name)
+
+        # Determine if the demand flexibility profile is indexed by zone, bus, or both
+        area_indicator = [1 if "zone." in x else 0 for x in df.columns]
+        if len(area_indicator) == sum(area_indicator):
+            # Demand flexibility profile only contains zones; prune accordingly
+            zone_id = sorted(self.grid.bus.zone_id.unique())
+            zone_id = [f"zone.{x}" for x in zone_id]
+            df = df.loc[:, df.columns.isin(zone_id)]
+        else:
+            bus_id = sorted(self.grid.bus.bus_id.unique())
+            bus_id = [str(x) for x in bus_id]
+            if len(area_indicator) > 0:
```

Comment:
Should this be `if sum(area_indicator) > 0`?

Reply:
Yes, I'll fix that.

Reply:
Related - I think this conditional block could be collapsed to 1 layer deep:

```
cols = []
if sum(area_indicator) > 0:
    # append zone_id to cols
if sum(area_indicator) < len(area_indicator):
    # append bus_id to cols
df = df.loc[:, df.columns.isin(cols)]
```

Reply:
Yeah, that's cleaner. Thanks! I'll be sure to implement that.

Reply:
Done.